slsim/lens.py

```diff
@@ -732,6 +732,12 @@ def _extended_single_source_magnification(self, source, source_index):
                 source_index
             ]
             kwargs_source_mag = kwargs_params["kwargs_source"][source_index]
+        if isinstance(light_model_list, list):
```

Comment:
I don't think we need those things as the components should already be defined as lists

Reply:
These lines are applicable while accessing keywords in some case (just one case where we have multisource and single sersic light profile. In this case above line will produce str and dictionary due to the indexing. So, need to convert them to list.)